go/libraries/doltcore/mvdata/table_data_loc.go

```diff
@@ -105,8 +105,12 @@ func (dl TableDataLocation) NewCreatingWriter(ctx context.Context, mvOpts DataMo
 	if err != nil {
 		return nil, err
 	}
+	updatedRoot, err := root.PutTable(ctx, dl.Name, tbl)
+	if err != nil {
+		return nil, err
+	}
 
-	tableEditor, err := doltdb.NewTableEditor(ctx, tbl, outSch)
+	tableEditor, err := doltdb.CreateTableEditSession(updatedRoot, doltdb.TableEditSessionProps{}).GetTableEditor(ctx, dl.Name, outSch)
```

Comment:
I'm heavily against that, as it can accidentally lead to devs using this `CreateSingleTableEditor` over creating a session and getting an editor from there. I want it to be clear and explicit that you need to get editors from a session, and that all of your editors need to come from the same session. If anything, I may need to separate this into two steps with an intermediate variable to make that more clear, but I'm against anything that may suggest you can create an editor without a session first.

Later on down the line, if the code is being edited and another table editor is added, it's easy to see two ways to make an editor and think "I only need this one" and call the single, even though a session exists further up the stack.

Reply:
Fine by me